### PR TITLE
Hive Progress is different than Presto’s

### PR DESCRIPTION
**Pull request by wyndhblb**
also a result can be “empty” but not none for returned results

### CODE REVIEW

File: superset/db_engine_specs.py [hunk 1]
```diff
@@ -675,18 +653,20 @@ def handle_cursor(cls, cursor, query, session):
             hive.ttypes.TOperationState.RUNNING_STATE,
         )
         polled = cursor.poll()
+        import logging
```
Comment (xrmx) [inline superset/db_engine_specs.py]: does not look the right place for an import

File: superset/db_engine_specs.py [hunk 2]
```diff
@@ -675,18 +653,20 @@ def handle_cursor(cls, cursor, query, session):
...
-            query = session.query(type(query)).filter_by(id=query.id)
+            query = session.query(type(query)).filter_by(id=query.id).one()
             if query.status == QueryStatus.STOPPED:
                 cursor.cancel()
                 break
-
             resp = cursor.fetch_logs()
-            if resp and resp.log:
-                progress = cls.progress(resp.log)
+            if resp and len(resp) > 0:
```
Comment (xrmx) [inline superset/db_engine_specs.py]: `if resp:` is enough

File: superset/db_engines/hive.py [hunk 3]
```diff
@@ -30,8 +30,9 @@ def fetch_logs(self, max_rows=1024,
             )
             response = self._connection.client.FetchResults(req)
             hive._check_status(response)
-            assert not (
-                response.results.rows, 'expected data in columnar format'
+            assert (
+                (not response.results.rows or len(response.results.rows) == 0),
```
Comment (xrmx) [inline superset/db_engines/hive.py]: Previous code was perfectly fine, the part you added is unneeded

**Review by xrmx — CHANGES_REQUESTED** [review]

**Comment by mistercrunch** [comment]
I think you remove the notion of stages. If a hive query has 10 stages, and we're half way through the last stage, it should show 95%-ish.

File: superset/db_engine_specs.py (hunk 1 above)
Comment (wyndhblb) [inline superset/db_engine_specs.py]: agreed ..

File: superset/db_engine_specs.py (hunk 2 above)
Comment (wyndhblb) [inline superset/db_engine_specs.py]: Just a "nulls" vs "things that define __len__" not begin the same .. so should `resp` return an object (that's not null, but empty) .. lots of folks forget the `__nonzero__` method is a thing.

File: superset/db_engines/hive.py (hunk 3 above)
Comment (wyndhblb) [inline superset/db_engines/hive.py]: Just a "nulls" vs "things that define __len__" not begin the same .. so should `resp` return an object (that's not null, but empty) .. lots of folks forget the `__nonzero__` method is a thing.

Comment (wyndhblb) [inline superset/db_engines/hive.py]: just a habit of mine, i can remove those things if it's a blocker for you.

File: superset/db_engine_specs.py (hunk 2 above)
Comment (xrmx) [inline superset/db_engine_specs.py]: AFAIU __len__ is called if __nonzero__ is not defined so should be covered too
https://docs.python.org/2/reference/datamodel.html#object.__nonzero__

**Pull request closed without merging**